### PR TITLE
feat: implement sign up component

### LINKED ISSUE #6: Create Sign up Component
## Description
Create a register component to enable sme/investors register on the platform

## Solution
A register component should be able to take in data from sme/investor, process registration and store via the API endpoint, validate the user and notify the user.

## Alternative
none

### PR DESCRIPTION
**Pull request by Yasholma**
- Created sign up component
- Created validation rules
- Created unit tests for validation function
- Created unit test for sign up component

Delivers #6

## Description
I created the sign up component so that the functionalities will be implemented. The sign up component leverages the generic validation file I also created will can be used somewhere else. There is instant validation for any input that is touched.

Delivers #6 

## How Has This Been Tested?

1. In your terminal, run git clone https://github.com/BuildForSDG/SmeVest-Frontend.git
2. Run cd SmeVest-Frontend
3. Run git fetch origin ft-signup-component
4. Run git checkout ft-signup-component
5. Run yarn install
6. Run yarn start
7. Go to http://localhost:1234/signup on your browser
8. Run yarn run test to run unit tests

## Screenshots (if applicable, else remove this line / section)
image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Alasco95 — DISMISSED** [review]

File: src/components/SignUp/SignUp.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.setState({ loading: true });
+  }
+
+  handleInputChange(e) {
+    const name = e.target;
+    const value = e.target;
```
Comment (oyewoas) [inline src/components/SignUp/SignUp.js]: Welldone @Yasholma , i wanted to try typing on the input fields to see how the validation works but the input fields were not updating with the values until i changed this line of code to `const { name } = e.target` and `const {value} = e.target` did you leave it out intentionally?, and  i think you can add `e.preventDefault()` also at the top of the `handleInputChange` method.

Comment (Yasholma) [reply src/components/SignUp/SignUp.js]: I destructured that initially, then codacy slapped me. I tried this current code it was working, maybe my browser cached it. But I repushed a new version again

Comment (oyewoas) [reply src/components/SignUp/SignUp.js]: Nice

**Review by oyewoas — APPROVED** [review]
Nice work @Yasholma , you can move to integrating with the signup API

**Review by Alasco95 — APPROVED** [review]

**Review by Johnny2020Ade — COMMENTED** [review]
Good progress @Yasholma

**Review by Johnny2020Ade — APPROVED** [review]

**Pull request merged**